wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java

```diff
@@ -71,12 +84,43 @@ public boolean isAvailable() {
   boolean isConnectionSuccessful(HttpURLConnection conn) {
     try {
       int responseCode = conn.getResponseCode();
-      return responseCode >= 200 && responseCode < 400;
+      boolean success = isConnectionSuccessful(responseCode);
+
+      if (!success && LOG.isDebugEnabled()) {
+        String response;
+        try {
+          InputStream in = conn.getErrorStream();
+          if (in == null) {
+            in = conn.getInputStream();
+          }
+
+          try (Reader reader = new InputStreamReader(in)) {
+            response = CharStreams.toString(reader);
+          }
+        } catch (Exception e) {
+          response = "failed to ready response: " + e.getMessage();
```

Comment:
do we need to re-throw `Exception` here ?

Reply:
Nope. It's server check where we just read the response and retry later when it fails.